cpp/src/IceStorm/Service.cpp

```diff
@@ -338,13 +338,13 @@ ServiceI::start(const string& name, const shared_ptr<Communicator>& communicator
                 // We're not using an IceGrid deployment. Here we need
                 // a proxy which is used to create proxies to the
                 // replicas later.
-                _managerProxy = uncheckedCast<TopicManagerPrx>(topicAdapter->createProxy(topicManagerId));
+                _managerProxy = TopicManagerPrx{topicAdapter->createProxy(topicManagerId)};
```

Comment:
We could add a type parameter to createProxy, add, etc., but since they return non-null proxies, it wouldn't be a big improvement:
```cpp
_managerProxy = TopicManagerPrx{topicAdapter->createProxy(topicManagerId)};
// vs
_managerProxy = topicAdapter->createProxy<TopicManagerPrx>(topicManagerId);
```
doesn't seem worth it.

It's a bigger improvement when the returned proxy can be null and is reasonably often null, like with propertyToProxy.